src/download.rs

```diff
@@ -248,19 +248,18 @@ pub fn download_file(
             copy_largebuf(&mut pb.wrap_read(response), &mut buffer)?;
             let downloaded_length = pb.position() - starting_length;
             pb.finish_and_clear();
-            let time_taken = starting_time.elapsed().as_secs();
-            if let Some(speed) = downloaded_length.checked_div(time_taken) {
+            let time_taken = starting_time.elapsed();
+            if time_taken >= Duration::from_micros(1) {
                 eprintln!(
                     "Done. {} in {} ({}/s)",
                     HumanBytes(downloaded_length),
-                    humantime::format_duration(Duration::from_secs(time_taken)),
-                    HumanBytes(speed)
+                    humantime::format_duration(
+                        Duration::from_micros(time_taken.as_micros() as u64)
+                    ),
```

Comment:
Can we pass `time_taken` directly to `format_duration()`?

```rust
humantime::format_duration(time_taken)
```

Reply:
Fixed.

Reply:
Fixed. Also, change the condition to `Duration` is non-zero.